pennylane/transforms/defer_measurements.py

```diff
@@ -405,15 +405,10 @@ def _add_control_gate(op, control_wires, reduce_postselected):
     for branch, value in items:
         if value:
             # Empty sampling branches can occur when using _postselected_items
-            if branch == ():
-                new_ops.append(op.base)
-                continue
-            qscript = qml.tape.make_qscript(
-                ctrl(
-                    lambda: qml.apply(op.base),  # pylint: disable=cell-var-from-loop
-                    control=Wires(control),
-                    control_values=branch,
-                )
-            )()
-            new_ops.extend(qscript.circuit)
+            new_op = (
+                op.base
+                if branch == ()
+                else ctrl(op.base, control=Wires(control), control_values=branch)
+            )
+            new_ops.append(new_op)
```

Comment:
Wait, what :thinking: Why did we go this convoluted way via a `callable` before?

And how come this changes now?

Reply:
I have no context for why we were using the old way. It was before my time 😅 . I had to change this because the old logic wasn't playing nice with program capture, and instead of controlled operators being queued, just the base operators were being queued.

Reply:
The nice thing about making this change, however, is that `defer_measurements` seems to be working with program capture as it should 😄

Reply:
Very good :) If no tests are breaking, the new easier logic should be fine, or we'll have to improve the tests soon :D